### PR TITLE
Task 187

### PR DESCRIPTION
**Pull request by emymaria**
#### Short Description

Link Builder Class - Link Builder class used hateoas. Deleted this class. It was used in a single method in link builder service,
					this method had usages only in link builder service test. So I assume it is safe to delete the class.
					All the tests both in core and web passed after deleting also. 

#### Proposed Changes

  * Deleted the LinkBuilder Class
  * Removed the method in LinkBuilderService that used the LinkBuilder Class
  *Removed 3 tests from LinkBuilderServiceTest class that used the deleted method in LinkBuilderService

**Fixes**: #

### CODE REVIEW

**Review by salmma — CHANGES_REQUESTED** [review]
- Please remove unused imports in LinkBuilderServiceTest
- Remove hateoas [Links](https://github.com/UST-QuAntiL/qc-atlas/blob/5fe2dfb084d0ec1f3087a15cd289c75934b9a9e3/org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/EntityModelConverter.java#L25)
- check if [HateoasSortHandlerMethodArgumentResolver](https://github.com/UST-QuAntiL/qc-atlas/blob/5fe2dfb084d0ec1f3087a15cd289c75934b9a9e3/org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/ListParametersMethodArgumentResolver.java#L45) can be removed

**Review by manuwei — CHANGES_REQUESTED** [review]
The removal of the LinkBuilder is fine :) 
Can you have at all imports with "hateoas" again? I think there are still imports/occurences in the code that can be removed, 
e.g., @Relation annotations and `addLinks(...)` we add hateoas links to an entity

**Comment by emymaria** [comment]
> * Please remove unused imports in [LinkBuilderServiceTest](https://github.com/UST-QuAntiL/qc-atlas/blob/9b44a89b7a9476fae7f96b86664abfe5a91a12a8/org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/linkassembler/LinkBuilderServiceTest.java)
> * Remove hateoas [Links](https://github.com/UST-QuAntiL/qc-atlas/blob/5fe2dfb084d0ec1f3087a15cd289c75934b9a9e3/org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/EntityModelConverter.java#L25)
> * check if [HateoasSortHandlerMethodArgumentResolver](https://github.com/UST-QuAntiL/qc-atlas/blob/5fe2dfb084d0ec1f3087a15cd289c75934b9a9e3/org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/ListParametersMethodArgumentResolver.java#L45) can be removed

**Comment by emymaria** [comment]
Hallo
I removed remove unused imports in LinkBuilderServiceTest.
Removed hateoas.Links - Links was used in EntityModelConverter class which was in turn used in swagger configuration. Removed that part of swagger configuration and tested the swagger to find it working fine. So I deleted EntityModelConverter and its test class. Also I find it safe to delete OverrideModelConverter class and its test class also.
remove @relation annotations and addLinks(...)  - After new build I could not find these keywords, so I assume it got removed

**Review by salmma — CHANGES_REQUESTED** [review]
- Please remove your out-commented code
- check if the [LinkRemoverModelController](https://github.com/UST-QuAntiL/qc-atlas/blob/5b357ebbfb5dab98da982a480d93412e2ab03a8f/org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/LinkRemoverModelConverter.java#L39) can be removed
- check if [OverrideModelConverter](https://github.com/UST-QuAntiL/qc-atlas/blob/5b357ebbfb5dab98da982a480d93412e2ab03a8f/org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/OverrideModelConverter.java#L40) can be removed
- check if further classes/methods can be removed in context of HATEOAS :)

File: org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/EntityModelConverter.java [hunk 1]
```diff
@@ -1,93 +0,0 @@
...
-import io.swagger.v3.core.util.Json;
-import io.swagger.v3.oas.models.media.Schema;
-
-/**
- * Custom converter for EntityModel classes.
- * <p>
- * Spring HATEOAS' EntityModel class is special because it uses Jackson's @JsonWrapped to "copy" the actual DTO's
- * properties into the EntityModel instance. Unfortunately, information on whether properties are required or not is
- * lost in the process. This wrapper aims to fix that by copying the DTO's schema and then manually adding the
- * EntityModel-specific fields (for now, just _links).
```
Comment (manuwei) [inline org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/utils/EntityModelConverter.java:42]: did you do some manual testing to check if the entitys are still generated/saved/outputted correctly?

File: org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/utils/EntityModelConverterTest.java [hunk 2]
```diff
@@ -1,61 +0,0 @@
...
-import org.junit.jupiter.api.Test;
-import org.springframework.hateoas.EntityModel;
-
-import io.swagger.v3.core.converter.AnnotatedType;
-import io.swagger.v3.core.converter.ModelConverters;
-import io.swagger.v3.core.util.Json;
-import lombok.Data;
-import lombok.NoArgsConstructor;
-
-public class EntityModelConverterTest {
```
Comment (manuwei) [inline org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/utils/EntityModelConverterTest.java:36]: although we do not add hateoas links to our entities anymore, I think we still need a test for entity model converting

**Review by manuwei — CHANGES_REQUESTED** [review]
I also added some further remarks about the changes :) can you have  a look at them?

**Comment by emymaria** [comment]
I find that the LinkRemoverModelController and OverrideModelConverter can be removed. These were used in swagger configuration. Since I removed certain configurations in swagger that use these deleted class, I am testing the swagger. It is in progress. I have checked first 10 apis. I wonder if I have to test every single one or are there any high priority ones that I can test.

**Review by salmma — CHANGES_REQUESTED** [review]
LG - however, please add documentation as comment on top of the LinkBuilderService class to clarify that this class is used to avoid hard-coded urls in all ControllerTests.

**Review by manuwei — APPROVED** [review]

**Comment by emymaria** [comment]
Added comments in LinkBuilderService on where the class is used

**Review by salmma — APPROVED** [review]
LGTM

**Pull request merged**